From 8bdf14986a1baee1d44724cc66bc24c55418ebf5 Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Wed, 7 Aug 2019 14:53:26 +0200
Subject: [PATCH] attr: mark nested attributes as NLA_F_NESTED

Kernel 5.2 is adding stricter checking for netlink messages.
In particular, for certain API it checks now that NLA_F_NESTED flag is
set for nested attributes ([1]).

Do like libmnl, which always adds this flag ([2]). So we should do
that as well.

[1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=b424e432e770d6dd572765459d5b6a96a19c5286
[2] https://git.netfilter.org/libmnl/tree/src/attr.c?id=5937dfcb0185f5cb9cf275992ea701ec4e619d9c#n535

Change-Id: I9012d820ed72b13d4562e144c48bc7003d621f09
---
 lib/attr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/attr.c b/lib/attr.c
index bdf3c0f..a76a284 100644
--- a/lib/attr.c
+++ b/lib/attr.c
@@ -796,7 +796,7 @@ struct nlattr *nla_nest_start(struct nl_msg *msg, int attrtype)
 {
 	struct nlattr *start = (struct nlattr *) nlmsg_tail(msg->nm_nlh);
 
-	if (nla_put(msg, attrtype, 0, NULL) < 0)
+	if (nla_put(msg, NLA_F_NESTED | attrtype, 0, NULL) < 0)
 		return NULL;
 
 	NL_DBG(2, "msg %p: attr <%p> %d: starting nesting\n",
-- 
2.7.4

